templates/tools/dockerfile/python_debian11.include

```diff
@@ -0,0 +1,53 @@
+FROM debian:bullseye
+  
+# Install Git and basic packages.
+RUN apt-get update && apt-get install -y ${'\\'}
+  autoconf ${'\\'}
+  autotools-dev ${'\\'}
+  build-essential ${'\\'}
+  bzip2 ${'\\'}
+  ccache ${'\\'}
+  curl ${'\\'}
+  dnsutils ${'\\'}
+  gcc ${'\\'}
+  gcc-multilib ${'\\'}
+  git ${'\\'}
+  golang ${'\\'}
+  gyp ${'\\'}
+  lcov ${'\\'}
+  libc6 ${'\\'}
+  libc6-dbg ${'\\'}
+  libc6-dev ${'\\'}
+  libgtest-dev ${'\\'}
+  libtool ${'\\'}
+  make ${'\\'}
+  perl ${'\\'}
+  strace ${'\\'}
+  telnet ${'\\'}
+  unzip ${'\\'}
+  wget ${'\\'}
+  zip && apt-get clean
+
+#================
+# Build profiling
+RUN apt-get update && apt-get install -y time && apt-get clean
+
+# Install Python 3.7 from source (and installed as a default python3)
+# (Bullseye comes with Python 3.9 which isn't supported by pytype yet)
+RUN apt update && apt install -y build-essential zlib1g-dev libncurses5-dev libgdbm-dev ${'\\'}
+                                 libnss3-dev libssl-dev libreadline-dev libffi-dev
+RUN curl -O https://www.python.org/ftp/python/3.7.9/Python-3.7.9.tar.xz && ${'\\'}
+    tar -xf Python-3.7.9.tar.xz && ${'\\'}
+    cd Python-3.7.9 && ${'\\'}
+    ./configure && ${'\\'}
+    make -j 4 && ${'\\'}
+    make install
+RUN curl https://bootstrap.pypa.io/get-pip.py | python3
+
+# Install Python 2.7
```

Comment:
do we know why exactly we still need python2.7. Since we're already doing updates, we can check whether we could get rid of python2.7 while we're at it.

Reply:
There are a couple of sanity scripts which are not ready to be used in Python3. If they're migrated, this can be switched to Python 3. We can do python3-migration in a separate PR.

Reply:
If you could look into converting all the remaining sanity checks into python3 and then got rid of the python2.7 that would be great. 
The nice things is that once python2.7 is gone from the sanity docker image, that will protect us against python3 incompatible scripts since the sanity checks excercise quite a few of our testing/misc scripts.